### PR TITLE
feat(eventbus): allow a handler to subscribe to multiple events

### PR DESCRIPTION
**Pull request by YdreamW**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)

`plugin/eventbus`  `eventbus-decorator`  `eventbus-runtime` 

##### Description of change

allow a handler to subscribe to multiple events

### CODE REVIEW

File: core/eventbus-decorator/README.md [hunk 1]
```diff
@@ -56,3 +56,29 @@ export class Foo {
...
+
+### inject event context 
+inject event context  if you want to know which event is being handled.
+The context param must be the first param
+
+```ts
+@Event('hello')
+@Event('hi')
+export class Foo {
+  async handle(@EventContext ctx: IEventContext, msg: string):Promise<void> {
```
Comment (killagu) [inline core/eventbus-decorator/README.md:79]: 应该是 @EventContext()

File: core/eventbus-decorator/src/EventInfoUtil.ts [hunk 2]
```diff
@@ -2,13 +2,42 @@ import { EggProtoImplClass, MetadataUtil } from '@eggjs/core-decorator';
...
   static getEventName(clazz: EggProtoImplClass): EventName | undefined {
-    return MetadataUtil.getMetaData(EVENT_NAME, clazz);
+    const eventNameList = MetadataUtil.initOwnArrayMetaData<EventName>(EVENT_NAME, clazz, []);
+    if (eventNameList.length !== 0) {
+      return eventNameList[eventNameList.length - 1];
+    }
+    return undefined;
+  }
+
+  static setEventHandlerContextInject(enable: boolean, clazz: EggProtoImplClass): void {
```
Comment (killagu) [inline core/eventbus-decorator/src/EventInfoUtil.ts:36]: 这里应该还要记录是第几个参数，注解不一定是在第一个的。

**Comment by killagu** [comment]
🎉实现的非常好。感谢。

File: core/eventbus-decorator/src/EventInfoUtil.ts (hunk 2 above)
Comment (YdreamW) [reply core/eventbus-decorator/src/EventInfoUtil.ts:36]: 修饰器里做了限制，只允许第一个位置注入 context。
主要考虑类型提示，如果允许在任意位置注入，类型不太好做提示。声明事件后，参数列表也就确定了。
```ts
declare module '@eggjs/eventbus-decorator' {
  interface Events {
    hello: (msg: string) => Promise<void>;
  }
}
```

**Comment by YdreamW** [comment]
deprecated 的接口没覆盖到，覆盖率掉了，我补下单测

File: core/eventbus-decorator/src/EventContext.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// use @eggjs/tegg as namespace
+// eslint-disable-next-line import/no-unresolved
+import { Events } from '@eggjs/tegg';
```
Comment (fengmk2) [inline core/eventbus-decorator/src/EventContext.ts:3]: @killagu @YdreamW 这里会引发循环依赖吗？

Comment (YdreamW) [reply core/eventbus-decorator/src/EventContext.ts:3]: 应该不会，这里引入的是 merge 后的 namespace。
https://www.typescriptlang.org/docs/handbook/declaration-merging.html

Comment (killagu) [reply core/eventbus-decorator/src/EventContext.ts:3]: 那应该加个 import type?

**Review by killagu — APPROVED** [review]
LGTM

**Comment by killagu** [comment]
- @eggjs/aop-decorator@3.29.0
 - @eggjs/controller-decorator@3.29.0
 - @eggjs/core-decorator@3.29.0
 - @eggjs/egg-module-common@3.29.0
 - @eggjs/eventbus-decorator@3.29.0
 - @eggjs/standalone-decorator@3.29.0
 - @eggjs/tegg@3.29.0
 - @eggjs/tegg-aop-plugin@3.29.0
 - @eggjs/tegg-aop-runtime@3.29.0
 - @eggjs/tegg-background-task@3.29.0
 - @eggjs/tegg-common-util@3.29.0
 - @eggjs/tegg-config@3.29.0
 - @eggjs/tegg-controller-plugin@3.29.0
 - @eggjs/tegg-dynamic-inject@3.29.0
 - @eggjs/tegg-dynamic-inject-runtime@3.29.0
 - @eggjs/tegg-eventbus-plugin@3.29.0
 - @eggjs/tegg-eventbus-runtime@3.29.0
 - @eggjs/tegg-lifecycle@3.29.0
 - @eggjs/tegg-loader@3.29.0
 - @eggjs/tegg-metadata@3.29.0
 - @eggjs/tegg-orm-decorator@3.29.0
 - @eggjs/tegg-orm-plugin@3.29.0
 - @eggjs/tegg-plugin@3.29.0
 - @eggjs/tegg-runtime@3.29.0
 - @eggjs/tegg-schedule-decorator@3.29.0
 - @eggjs/tegg-schedule-plugin@3.29.0
 - @eggjs/tegg-standalone@3.29.0
 - @eggjs/tegg-transaction-decorator@3.29.0

**Pull request merged**